packages/services/storage/src/index.ts

```diff
@@ -3686,34 +3686,61 @@ export async function createStorage(connection: string, maximumPoolSize: number)
     },
 
     async purgeExpiredSchemaChecks(args) {
-      await pool.query(sql`
-        DELETE
-        FROM
-          "public"."schema_checks"
-        WHERE
-          "expires_at" <= ${args.expiresAt.toISOString()}
-      `);
+      return await pool.transaction(async pool => {
+        const result = await pool.any<unknown>(sql`
+          DELETE
+          FROM "public"."schema_checks"
+          WHERE 
+            "id" = ANY(
+              SELECT
+                "id"
+              FROM
+                "public"."schema_checks"
+              WHERE
+                "expires_at" <= ${args.expiresAt.toISOString()}
+              LIMIT
+                1000
+            )
+          RETURNING
+            "schema_sdl_store_id" as "id1",
+            "supergraph_sdl_store_id" as "id2",
+            "composite_schema_sdl_store_id" as "id3"
+        `);
+        const ids = PurgeExpiredSchemaChecksIDModel.parse(result);
 
-      return 0;
-    },
+        if (ids.size === 0) {
+          return {
+            deletedSchemaCheckCount: result.length,
+            deletedSdlStoreCount: 0,
+          };
+        }
+        const deletedRecords = await pool.any<unknown>(sql`
+          DELETE
+          FROM
+            "sdl_store"
+          WHERE
+            "id" = ANY(
+              ${sql.array(Array.from(ids), 'text')}
+            )
+            AND NOT EXISTS (
+              SELECT
+                1
+              FROM
+                "schema_checks"
+              WHERE
+                "schema_checks"."schema_sdl_store_id" = "sdl_store"."id"
+                OR "schema_checks"."composite_schema_sdl_store_id" = "sdl_store"."id"
+                OR "schema_checks"."supergraph_sdl_store_id" = "sdl_store"."id"
+            )
+          RETURNING
+            true as "d"
+        `);
 
-    async purgeUnusedSchemasInStore() {
-      await pool.query(sql`
-        DELETE FROM "public"."sdl_store"
-        WHERE id NOT IN (
-            SELECT DISTINCT schema_sdl_store_id
-            FROM "public"."schema_checks"
-            WHERE schema_sdl_store_id IS NOT NULL
-          UNION
-            SELECT DISTINCT supergraph_sdl_store_id
-            FROM "public"."schema_checks"
-            WHERE supergraph_sdl_store_id IS NOT NULL
-          UNION
-            SELECT DISTINCT composite_schema_sdl_store_id
-            FROM "public"."schema_checks"
-            WHERE composite_schema_sdl_store_id IS NOT NULL
-        )
-      `);
+        return {
+          deletedSchemaCheckCount: result.length,
```

Comment:
```suggestion
          deletedSchemaCheckCount: result.size,
```

I think it should be size